### PR TITLE
Query array fix

### PR DESCRIPTION
**Pull request by rjt-gupta**
Fix for #3072 and #3254 

Added `format_pair` function and changed the implementation of `format_dict`.
Did relevant changes in urlencoded.py

### CODE REVIEW

**Comment by rjt-gupta** [comment]
This is ready for review.

File: mitmproxy/contentviews/base.py [hunk 1]
```diff
@@ -35,6 +35,35 @@ def __call__(self, data: bytes, **metadata) -> TViewResult:
         raise NotImplementedError()  # pragma: no cover
 
 
+def format_pair(item: TViewLine) -> typing.Iterator[TViewLine]:
```
Comment (mhils) [inline mitmproxy/contentviews/base.py]: This method should be `format_pairs` and not `format_pair`, as it formats multiple (key, value) pairs.

Comment (mhils) [reply mitmproxy/contentviews/base.py]: Second, `item` is a bad name for a list. `items` would work, or `pairs`, or something like that.

Comment (mhils) [reply mitmproxy/contentviews/base.py]: Also, the TViewLine type is used to indicate `(style, text)` tuples for rendering. That's not really what we have as input here, which should just be an `Iterable[Tuple[TTextType,TTextType]]`.

File: mitmproxy/contentviews/base.py [hunk 2]
```diff
@@ -35,6 +35,35 @@ def __call__(self, data: bytes, **metadata) -> TViewResult:
...
+        ("key", key    )
+        ("value", value)
+    ]
+    where key is padded to a uniform width
+    """
+
+    max_key_len = max((len(k[0]) for k in item), default=0)
+    max_key_len = min((max_key_len, KEY_MAX), default=0)
+
+    for k in item:
```
Comment (mhils) [inline mitmproxy/contentviews/base.py]: We can make this much more readable by using `for key, value in pairs`. :-)

File: mitmproxy/contentviews/base.py [hunk 3]
```diff
@@ -35,6 +35,35 @@ def __call__(self, data: bytes, **metadata) -> TViewResult:
...
+        if isinstance(k[0], bytes):
+
+            temp = b"k[0]" + b":"
```
Comment (mhils) [inline mitmproxy/contentviews/base.py]: That's definitely wrong.

File: mitmproxy/contentviews/base.py [hunk 4]
```diff
@@ -47,18 +76,7 @@ def format_dict(
...
-        if isinstance(key, bytes):
-            key += b":"
-        else:
-            key += ":"
-        key = key.ljust(max_key_len + 2)
-        yield [
-            ("header", key),
-            ("text", value)
-        ]
+    return format_pair(list(d.items()))
```
Comment (mhils) [inline mitmproxy/contentviews/base.py]: There's no need to wrap `d.items()` in `list()`.

File: mitmproxy/contentviews/base.py (hunk 1 above)
Comment (rjt-gupta) [reply mitmproxy/contentviews/base.py]: okay!

Comment (rjt-gupta) [reply mitmproxy/contentviews/base.py]: Sorry i misinterpreted TViewLine type.

File: mitmproxy/contentviews/base.py (hunk 3 above)
Comment (rjt-gupta) [reply mitmproxy/contentviews/base.py]: Sorry about this.

**Comment by rjt-gupta** [comment]
It was a learning experience.Thanks!
I will make all the changes.

**Pull request merged**